x/storage/keeper/keeper.go

```diff
@@ -2230,13 +2230,21 @@ func (k Keeper) SetTag(ctx sdk.Context, operator sdk.AccAddress, grn types2.GRN,
 		if !found {
 			return types.ErrNoSuchBucket.Wrapf("bucketName: %s", bucketName)
 		}
-		resOwner := sdk.MustAccAddressFromHex(bucketInfo.Owner)
-		if !operator.Equals(resOwner) {
-			return types.ErrAccessDenied.Wrapf(
-				"Only resource owner can set tag, operator (%s), owner(%s)",
-				operator.String(), resOwner.String())
+		if ctx.IsUpgraded(upgradetypes.HulunbeierPatch) {
+			// check permission
+			effect := k.VerifyBucketPermission(ctx, bucketInfo, operator, permtypes.ACTION_UPDATE_BUCKET_INFO, nil)
```

Comment:
Do we need to update `VerifyBucketPermission` function?
I am not getting how it works - we only add a new action type, however we do not change the verify logic for this action. Does it mean there is a default permission?

Reply:
The func will retrieve the specified policy and iterate all statements to verify if `permtypes.ACTION_UPDATE_BUCKET_INFO` we passed belongs any of actions of those statements.